tsconfig.json

```diff
@@ -2,18 +2,29 @@
   "files": ["globals.d.ts"],
   "compilerOptions": {
     "allowJs": true,
-    "declaration": true,
+    "allowSyntheticDefaultImports": true,
     "esModuleInterop": true,
     "forceConsistentCasingInFileNames": true,
     "isolatedModules": true,
     "jsx": "react-jsx",
     "lib": ["DOM", "DOM.Iterable", "ES2019"],
+    "module": "ESNext",
     "moduleResolution": "node",
-    "module": "ES2022",
-    "outDir": "dist",
     "resolveJsonModule": true,
+    "skipLibCheck": true,
     "strict": true,
     "target": "ES2019",
-    "types": ["vitest/globals"]
-  }
+    "types": ["vitest/globals"],
+    "downlevelIteration": true
+  },
+  "include": ["src/**/*.tsx", "src/**/*.ts", "src/**/*.jsx", "src/**/*.js"],
+  "exclude": [
+    "src/**/*.test.**",
+    "src/**/*.spec.**",
+    "src/**/*.stories.**",
+    "dist",
+    "vitest-setup.ts",
+    "vite.config.ts",
+    "vite.config.ts"
```

Comment:
The reason that we had a dev + production tsconfig before was that I can now introduce type errors in tests and stories. While VSCode does highlight them to me, `pnpm run type-check` does not because the tsconfig tells typescript to ignore these files.

Reply:
I'm not sure this is true. I can introduce a type error on main in one of the stories and it's not picked up by type-check.

<img width="1543" alt="image" src="https://github.com/user-attachments/assets/7da95e26-0e30-4761-ae52-10fbe37f783d" />

If I actually implement what you are talking about, we have 141 existing issues.

<img width="1543" alt="image" src="https://github.com/user-attachments/assets/c0a91723-1fff-4869-b0f6-e6a6791a88df" />

Reply:
I propose a separate ticket to cover this existing issue so we don't hold up the start of the DATA migration. If it was just a couple of errors we could absorb it now but 141 is not a 5 minute job. I'm happy to look into it once the DATA work is underway.

Reply:
https://sce.myjetbrains.com/youtrack/issue/VE-290/Type-checks-for-tests-and-stories-in-RUI